specs/builder.go

```diff
@@ -0,0 +1,312 @@
+					dummyField.PrefixLength = len(lengthStr)
+				}
+			}
+			if spec.Pad != nil {
+				pad := padDummy{}
+				switch reflect.TypeOf(spec.Pad).Elem().String() {
+				case "padding.leftPadder":
+					if spec.Pad.Inspect() != nil {
+						pad.Type = reflect.TypeOf(spec.Pad).Elem().String()
+						pad.Pad, _ = utf8.DecodeRune(spec.Pad.Inspect())
+					}
+				}
+				if pad.Type != "" {
+					dummyField.Padding = &pad
+				}
+			}
+
+		default:
+			continue
+		}
+
+		dummyJson.Fields[dummyName] = dummyField
+	}
+
+	outputBuffer := new(bytes.Buffer)
+	enc := json.NewEncoder(outputBuffer)
+	enc.SetEscapeHTML(false)
+	enc.SetIndent("", "\t")
+	err := enc.Encode(dummyJson)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to export message spec")
+	}
+
+	return outputBuffer.Bytes(), nil
+}
+
+func getPrefixInterface(prefixName string, length int) prefix.Prefixer {
+
+	prefixName = "*" + prefixName
+	var pref prefix.Prefixer
+
+	switch prefixName {
+	case reflect.TypeOf(prefix.ASCII.Fixed).String():
+		pref = prefix.ASCII.Fixed
+	case reflect.TypeOf(prefix.BCD.Fixed).String():
+		pref = prefix.BCD.Fixed
+	case reflect.TypeOf(prefix.Hex.Fixed).String():
+		pref = prefix.Hex.Fixed
+	case reflect.TypeOf(prefix.Binary.Fixed).String():
+		pref = prefix.Binary.Fixed
+	case reflect.TypeOf(prefix.EBCDIC.Fixed).String():
+		pref = prefix.EBCDIC.Fixed
+	}
+
+	if pref != nil {
+		return pref
+	}
+
+	return getVarPrefixInterface(prefixName, length)
+}
+
+func getVarPrefixInterface(prefixName string, length int) prefix.Prefixer {
+	var pref prefix.Prefixer
+
+	switch prefixName {
+	case reflect.TypeOf(prefix.ASCII.L).String():
+		if length == 1 {
+			pref = prefix.ASCII.L
+		} else if length == 2 {
+			pref = prefix.ASCII.LL
+		} else if length == 3 {
+			pref = prefix.ASCII.LLL
+		} else if length == 4 {
+			pref = prefix.ASCII.LLLL
+		}
+	case reflect.TypeOf(prefix.BCD.L).String():
+		if length == 1 {
+			pref = prefix.BCD.L
+		} else if length == 2 {
+			pref = prefix.BCD.LL
+		} else if length == 3 {
+			pref = prefix.BCD.LLL
+		} else if length == 4 {
+			pref = prefix.BCD.LLLL
+		}
+	case reflect.TypeOf(prefix.Hex.L).String():
+		if length == 1 {
+			pref = prefix.Hex.L
+		} else if length == 2 {
+			pref = prefix.Hex.LL
+		} else if length == 3 {
+			pref = prefix.Hex.LLL
+		} else if length == 4 {
+			pref = prefix.Hex.LLLL
+		}
+	case reflect.TypeOf(prefix.Binary.L).String():
+		if length == 1 {
+			pref = prefix.Binary.L
+		} else if length == 2 {
+			pref = prefix.Binary.LL
+		} else if length == 3 {
+			pref = prefix.Binary.LLL
+		} else if length == 4 {
+			pref = prefix.Binary.LLLL
+		}
+	case reflect.TypeOf(prefix.EBCDIC.L).String():
+		if length == 1 {
+			pref = prefix.EBCDIC.L
+		} else if length == 2 {
+			pref = prefix.EBCDIC.LL
+		} else if length == 3 {
+			pref = prefix.EBCDIC.LLL
+		} else if length == 4 {
+			pref = prefix.EBCDIC.LLLL
+		}
+	}
+
+	return pref
+}
+func getEncodingInterface(encName string) encoding.Encoder {
```

Comment:
we can use a map to get encoding as well.

Or (crazy idea), we can try to create encoding using reflect package if we know its name. Kind of: `enc := eval("package.${encName}")` (not sure how to do it).

Reply:
it seems that the approach isn't fit for var types with length. So used resource map

Reply:
let's move `var encodings` out of the `func`